cloudpathlib/cloudpath.py

```diff
@@ -759,6 +759,9 @@ def suffix(self) -> str:
     def suffixes(self) -> List[str]:
         return self._dispatch_to_path("suffixes")
 
+    def with_stem(self, stem: str) -> Self:
+        return self._dispatch_to_path("with_stem", stem)
```

Comment:
CPython implementation is trivial, so maybe we put this in a try-except and fallback to our backport of their implementation if it is not present:
https://github.com/python/cpython/pull/19295/files#diff-fa525485738fc33d05b06c159172ff1f319c26e88d8c6bb39f7dbaae4dc4105cR859-R862

Reply:
(this means that we won't need to skip the test or make a note about compatible versions)